balancergroup: improve observability around balancer cache behavior

easwars opened the pull request:
RELEASE NOTES: none

dfawley reviewed (commented):
Should these be at a higher verbosity?  I don't think we want to log these by default for everyone, do we?

easwars commented:
> Should these be at a higher verbosity? I don't think we want to log these by default for everyone, do we?

Fair point. Moved them under `V2` and also fixed one outlier log which had a `balancer group` prefix. We are already using a prefix logger here.

dfawley commented on internal/balancergroup/balancergroup.go:
```diff
@@ -403,7 +408,7 @@ func (bg *BalancerGroup) Remove(id string) {
 	sbToRemove, ok := bg.idToBalancerConfig[id]
 	if !ok {
-		bg.logger.Infof("balancer group: trying to remove a non-existing locality from balancer group: %v", id)
+		bg.logger.Warningf("Child policy for locality %q does not exist in the balancer group", id)
```
Should this even be `Error` so our tests fail?  I.e. does it imply a coding error on our part?

dfawley reviewed (approved):
Approving but please see the inline question.
